src/transformers/trainer.py

```diff
@@ -3493,7 +3530,7 @@ def _push_from_checkpoint(self, checkpoint_folder):
         if not self.is_world_process_zero() or self.args.hub_strategy == HubStrategy.END:
             return
         # If we haven't finished the last push, we don't do this one.
-        if self.push_in_progress is not None and not self.push_in_progress.is_done:
+        if self.push_in_progress is not None and not self.push_in_progress.is_done():
```

Comment:
I wonder if we still need to whole `push_in_progress` logic. It was necessary previously for the following workflow:
- Commit is done
- Start the push -> push a single commit
- Another commit is done
- Start to push -> if the previous push was not finished, then you push the two commits at the same time (and all the accompanying LFS files)

In that workflow you would therefore push the same LFS files twice, requiring way too much bandwidth.

But I wonder how that applies here, given that if you don't push a checkpoint here, then you're essentially skipping a commit?

Reply:
Added the argument to push everything (False by default) but we skip a commit if there is a push in progress yes. This was also the case before as well.

We still need an object to track the push_in_progress and know if it is finished or not.